src/smt/theory_str_mc.cpp

```diff
@@ -916,7 +909,7 @@ namespace smt {
 
         smt_params subsolver_params;
         smt::kernel subsolver(m, subsolver_params);
-        subsolver.set_logic(symbol("QF_BV"));
+        subsolver.set_logic(symbol("QF_CHAR"));
```

Comment:
You can do "QF_S"

Reply:
This is giving me trouble for some reason even after adding `setup_char()` and `setup_QF_BV()` to `setup_str()` in `smt_setup`. I'm taking a look

Reply:
Even doing `subsolver_params.m_string_solver=symbol("seq")` does not give me a working character theory

Reply:
For some reason if a string solver theory plugin is loaded at the same time as `theory_char` this does not work. I tried making a case for a "char" string solver in `setup_QF_S()` that only loads QF_BV and theory_char, and this works perfectly -- would this solution be acceptable since it doesn't create an ad_hoc `QF_CHAR` as before?